### PR TITLE
fix(cli): preserve root answer before /btw follow-up

### LINKED ISSUE #5001: CLI /btw can hide the completed main answer from compact scrollback
## Problem

In compact `texra chat`, submitting `/btw` while the root assistant response is still running can advance into the follow-up turn before the completed root assistant response is committed to the static transcript. The root answer is saved in `conversation.json`, but it is missing from raw terminal output/scrollback, so the user only sees the queued follow-up and the side answer.

## Reproduction

Run a compact PTY session:

```sh
texra chat --agent chat --model deepseekT --api-mode included
```

1. Submit a main prompt that asks for a marker such as `MAIN-ANSWER-CHECK`.
2. While the main response is running, submit `/btw` with a second marker such as `SIDE-ANSWER-CHECK`.
3. Inspect raw terminal output and the stored `conversation.json`.

Observed before the fix:

- raw terminal output showed the prompt, queued follow-up, and side answer
- raw terminal output did not contain the completed main answer
- `conversation.json` did contain the main assistant message

Evidence from the dogfood run:

- hidden-main raw transcript: `/var/folders/d1/qpfb8kqs3dj482nzfg_0yvkc0000gn/T/texra-live-chat-btw-post4999/05-complete.txt`
- stored conversation with main answer: `~/.texra/workspace-storage/texra-live-btw-work-zZnpFp-ea1bf718/executions/5bb1af4f2a82/conversation.json`

## Expected

The completed root assistant answer should be printed in terminal scrollback before the `/btw` follow-up turn starts, even in compact layouts.

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

- finalize/sync the completed root assistant turn at the tool-use `WAITING` boundary
- keep finalized static transcript entries in compact scrollback instead of dropping them under the compact row budget
- preserve immediate runtime follow-up queueing so `/btw` is visible to idle-continuation logic
- dedupe `waiting:`/`final:` synthetic fallbacks when the stream log never materializes the same assistant reply

Closes #5001.

## Validation

- `corepack pnpm vitest run src/test-kernel/cli/TuiStateAndFocus.vitest.mts src/test-kernel/cli/ConversationPane.vitest.mts`
- `corepack pnpm --filter @texra-ai/cli build`
- live compact PTY `/btw` marker check: main answer present, side answer present, main before side (`/var/folders/d1/qpfb8kqs3dj482nzfg_0yvkc0000gn/T/texra-live-btw-reviewfix2-pty-OClMND/clean.txt`)
- `node packages/cli/scripts/validate-tui.mjs --snapshot-dir /tmp/texra-cli-btw-dedupe-fix` (`all 82 scenarios passed`)
- `npm run format`
- `npm run lint` (0 errors; existing unrelated import-order warnings)

---

> [!NOTE]
> **Low Risk**
> Changes are confined to CLI TUI transcript timing, static scrollback budgeting, and dedupe logic; no auth, persistence, or agent execution contract changes beyond a new WAITING hook callback.
> 
> **Overview**
> Fixes the CLI chat TUI so the **root assistant answer stays visible** when the user sends a **`/btw` follow-up** while the run is in **WAITING** (e.g. after tools finish).
> 
> At the **tool-use WAITING boundary**, the chat runner now **syncs the stream log**, **adds a missing assistant line** from `lastResponse`, and **finalizes** transcript entries so the completed turn is promoted before idle follow-up handling runs.
> 
> **Static scrollback** no longer drops finalized conversation lines when the terminal uses a **compact row budget**—only the session header stays compact; entries are still appended in order.
> 
> **Deduping** for synthetic assistant fallbacks was tightened so a **waiting** snapshot and a later **final** fallback for the same text do not create two rows.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7a1217999dd307462a17d181984ca6b68af5278e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/cli/src/chat/tui/runChatTui.tsx [hunk 1]
```diff
@@ -1110,6 +1127,19 @@ export async function runChat(
         followUpTarget = session.streamId;
       }
       if (!followUpTarget || session.stopRequested) return;
+      if (!childFollowUpTarget) {
+        while (
+          !chatTuiCanDispatchRootFollowUp(followUpTarget) &&
+          !session.stopRequested &&
+          !session.runCompleted
+        ) {
+          await new Promise<void>((resolve) => setTimeout(resolve, 25));
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/runChatTui.tsx]: Addressed in 0585bee9e. Root follow-ups are now sent into the runtime queue as soon as the stream id exists; the TUI no longer waits for `WAITING` before calling `sendFollowUp`. The completed root turn is still finalized by `onBeforeWaiting`, so `session.hasQueuedFollowUp()` remains visible to idle-continuation checks while scrollback still gets the completed root answer before the follow-up turn.

File: packages/cli/src/chat/tui/runChatTui.tsx [hunk 2]
```diff
@@ -1110,6 +1127,19 @@ export async function runChat(
...
+        }
+        if (session.stopRequested || session.runCompleted) return;
+        syncStreamLog(followUpTarget);
+        finalizeAssistantTranscriptEntries(followUpTarget);
+        await new Promise<void>((resolve) => setTimeout(resolve, 50));
+      }
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/runChatTui.tsx]: Addressed in 0585bee9e. I removed the fixed 50ms delay entirely; after the stream-id wait, the code rechecks `stopRequested` once and immediately calls `sendFollowUp`, so there is no post-check delay window for a stopped run to receive the queued follow-up.

File: packages/cli/src/chat/tui/runChatTui.tsx [hunk 3]
```diff
@@ -1110,6 +1127,19 @@ export async function runChat(
         followUpTarget = session.streamId;
       }
       if (!followUpTarget || session.stopRequested) return;
+      if (!childFollowUpTarget) {
+        while (
+          !chatTuiCanDispatchRootFollowUp(followUpTarget) &&
+          !session.stopRequested &&
+          !session.runCompleted
+        ) {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/runChatTui.tsx]: Addressed in 0585bee9e. The new TUI-side wait loop was removed, so there is no long-running queue task that can spin across `/clear` or dispatch into a later run. Follow-up dispatch is back to the existing immediate runtime queue path; the new fix point is the `onBeforeWaiting` transcript finalization hook.

File: packages/cli/src/chat/tui/runChatTui.tsx [hunk 4]
```diff
@@ -962,6 +963,16 @@ export async function runChat(
...
+          onBeforeWaiting: (lastResponse) => {
+            if (!session.streamId) return;
+            syncStreamLog(session.streamId);
+            appendAssistantTranscriptIfMissing(
+              session.streamId,
+              lastResponse,
+              `waiting:${executionId}:${waitingTurn++}`,
+            );
+            finalizeAssistantTranscriptEntries(session.streamId);
+          },
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/runChatTui.tsx:975]: Addressed in 7a1217999. `appendAssistantTranscriptIfMissing` now dedupes synthetic final fallback rows that have the same normalized text and the same stream-log sequence boundary, so a `waiting:` fallback and a later `final:` fallback for the same missing log response collapse to one row. I also added a regression test for this exact waiting/final duplicate case while keeping the existing test that repeated identical replies in distinct turns remain visible.

File: packages/cli/src/chat/tui/state/transcript.ts [hunk 5]
```diff
@@ -51,6 +51,11 @@ export function appendAssistantTranscriptIfMissing(
       (entry) =>
         entry.id === entryId ||
         (!entry.synthetic &&
+          entry.role === 'assistant' &&
+          normalizeTranscriptText(entry.text) === normalized) ||
+        (entry.synthetic &&
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/state/transcript.ts:56]: Leaving this as-is deliberately: `local` and `process` synthetics are not assistant final responses, and broadening this dedupe would risk suppressing a legitimate later assistant answer that happens to share text with a local/process marker. The regression target is specifically the same assistant response being materialized as both `waiting:` and `final:` at the same `syntheticAfterSeq`.
